From bfc6e1217265c8c77a8382f5c7970d049e958250 Mon Sep 17 00:00:00 2001
From: Dan Elkouby <streetwalkermc@gmail.com>
Date: Sun, 30 Jul 2017 18:49:42 +0300
Subject: [PATCH] Create a new split container when switching a workspace
 container to split layout

The behavior before 52ce8c8 was to do it regardless of what layout we're
switching to.

Fixes #2846
---
 src/con.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/con.c b/src/con.c
index d346b9f45..94bfa6e9e 100644
--- src/con.c.orig
+++ src/con.c
@@ -1719,7 +1719,7 @@ void con_set_layout(Con *con, layout_t layout) {
             con->workspace_layout = ws_layout;
             DLOG("Setting layout to %d\n", layout);
             con->layout = layout;
-        } else if (layout == L_STACKED || layout == L_TABBED) {
+        } else if (layout == L_STACKED || layout == L_TABBED || layout == L_SPLITV || layout == L_SPLITH) {
             DLOG("Creating new split container\n");
             /* 1: create a new split container */
             Con *new = con_new(NULL, NULL);
